File: SaveCur/app/src/main/java/com/sy43/savecur/HomeActivity.java
package com.sy43.savecur;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.GridView;

import java.util.ArrayList;
import java.util.List;

public class HomeActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        GridView categoriesGridView = (GridView) findViewById(R.id.categories_grid_view);
        categoriesGridView.setAdapter(new CategoriesGridAdapter(this, getCategories()));
    }

    private List<Category> getCategories() {
        List<Category> categories = new ArrayList<Category>();
        categories.add(new Category("Food",150,0));
        categories.add(new Category("Party",80,15));
        categories.add(new Category("Other",180,70));
        categories.add(new Category("Other2",30,25));
        categories.add(new Category("Other3",90,16));
        categories.add(new Category("Other3",90,16));
        categories.add(new Category("Other3",90,16));
        categories.add(new Category("Other3",90,16));
        categories.add(new Category("Other3",90,16));
        categories.add(new Category("Other3",90,16));
        categories.add(new Category("Other3",90,16));

        return categories;
    }
}

File: Fluz/app/src/main/java/com/example/fluz/ui/HomeActivity.kt
package com.example.fluz.ui

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.AttributeSet
import android.view.Menu
import android.view.MenuItem
import android.view.View
import androidx.navigation.findNavController
import androidx.navigation.fragment.findNavController
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupWithNavController
import com.example.fluz.R
import com.example.fluz.databinding.ActivityHomeBinding
import com.example.fluz.databinding.ActivityMainBinding
import com.google.android.material.bottomappbar.BottomAppBar
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.google.android.material.floatingactionbutton.FloatingActionButton

class HomeActivity : AppCompatActivity() {
    private lateinit var binding: ActivityHomeBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityHomeBinding.inflate(layoutInflater)
        setContentView(binding.root)

        //Initialize the bottom navigation view
        //create bottom navigation view object
        val bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottomNavigationView)
        val navController = findNavController(R.id.nav_host_fragment_content_home_main)
        bottomNavigationView.setupWithNavController(navController)

        val bar = findViewById<BottomAppBar>(R.id.bottomAppBar)
        val fab = findViewById<FloatingActionButton>(R.id.btn_add)

        navController.addOnDestinationChangedListener { _, destination, _ ->
            if(destination.id == R.id.nav_transaction || destination.id == R.id.categories) {

                bottomNavigationView.visibility = View.GONE
                bar.visibility = View.GONE
                fab.hide()
            } else {

                bottomNavigationView.visibility = View.VISIBLE
                bar.visibility = View.VISIBLE
                fab.show()
            }
        }

        binding.btnAdd.setOnClickListener {
            navController.navigate(R.id.nav_transaction)
        }
    }


}